modules/ROOT/pages/api-gateway-encryption-mule4.adoc

```diff
@@ -0,0 +1,270 @@
++
+[CAUTION]
+--
+If you are changing from `FULL` to `SENSITIVE_ONLY`, any parameter not marked as `sensitive` will be shown as plain text in your filesystem.
+--
+
+== Policies That Support Encryption
+
+Below is a list of all policies that support encryption in their values, along with their minimum required version to support this feature. +
+Earlier versions of these policies, and policies not listed below will not encrypt the policies values even when the gateway capability detects an encryption key in its configuration.
+
+[%header%autowidth.spread,cols="a,a"]
+|===
+| Policy | Version
+| Basic Authentication | 1.2.0 and later.
+| Client ID Enforcement | 1.2.0 and later.
+| Header Injection | 1.1.0 and later.
+| IP Whitelist | 1.2.0 and later.
+| IP Blacklist | 1.2.0 and later.
+| JWT Validation | 1.1.0 and later.
+| Open AM | 1.2.0 and later.
+| Open ID | 1.2.0 and later.
+| OAuth 2.0 | 1.2.0 and later.
+| Ping Federate | 1.2.0 and later.
+|===
+
+== Custom Policy Encryption
+
+The recommended way of creating a new custom policy project is using the Maven archetype. +
+When creating a new custom policy project using the archetype, you can pass the `-DencryptionSupported=true` property to enable encryption in your custom policy. +
+See xref:custom-policy-getting-started.adoc#setting-up-a-project-with-the-archetype[Setting up a project with the archetype] for more detailed steps to do this.
+
+If you are developing a custom policy manually, or want to add encryption to an existing policy, follow the steps below:
+
+. Add the required dependencies in the policy `pom.xml` file. +
+You need to add the *Secure Configuration Property Module* dependency in your `pom.xml` file:
++
+[source,XML,linenums]
+--
+<dependency>
+   <groupId>com.mulesoft.modules</groupId>
+   <artifactId>mule-secure-configuration-property-module</artifactId>
+   <version>${securePropertyModuleVersion}</version>
+   <classifier>mule-plugin</classifier>
+</dependency>
+--
+. Add the *Secure Properties Element* to your policy's template file. +
+Paste the following snippet in your policy's XML template file, after your `<mule ...>` element:
++
+[source,XML,linenums]
+--
+{{#if encrypted}}
+<secure-properties:config key="${anypoint.platform.encryption_key}" file="${encryptedPropertiesFile}" name="encryptionConfiguration">
+   <secure-properties:encrypt algorithm="AES" mode="CBC"/>
+</secure-properties:config>
+{{/if}}
+--
++
+This feature supports only the AES Algorithm in CBC mode. +
+The gateway capability takes the `anypoint.platform.encryption_key` and `encryptedPropertiesFile` values from the system properties that you set at startup time for the gateway.
+. Define the sensitive values in the policy's YAML file. +
+By default, the values in a custom policy are not sensitive. You can define whether a value is sensitive or not in the YAML file describing your policy you can define which values are sensitive using the boolean element `sensitive`:
```

Comment:
"describing  your policy you can define" > I think these are two separate sentences "describing  your policy. You can define..." Please check.